File: backend/stockmate-backend/stockmate-backend/src/main/java/com/stockmate/stockmate_backend/dto/CreateProductRequest.java
package com.stockmate.stockmate_backend.dto;

import com.stockmate.stockmate_backend.entity.ProductCategory;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.math.BigDecimal;

@Data
public class CreateProductRequest {

    @NotBlank(message = "Name is required")
    private String name;

    @NotNull(message = "Category is required")
    private ProductCategory category;

    private String size;

    @NotNull(message = "Price is required")
    @DecimalMin(value = "0.0", inclusive = false, message = "Price must be greater than zero")
    private BigDecimal price;

    @NotNull(message = "Quantity is required")
    @Min(value = 0, message = "Quantity cannot be negative")
    private Integer quantity;

    private Integer lowStockThreshold;
}

File: android/app/src/main/java/edu/cit/cortes/stockmate/model/CreateProductRequest.kt
package edu.cit.cortes.stockmate.model

import com.google.gson.annotations.SerializedName

data class CreateProductRequest(
    @SerializedName("name") val name: String,
    @SerializedName("size") val size: String?,
    @SerializedName("category") val category: String,
    @SerializedName("price") val price: Double,
    @SerializedName("quantity") val quantity: Int,
    @SerializedName("lowStockThreshold") val lowStockThreshold: Int
)
